Fix: add an extra check in the DOM processing code for file tree nodes

EnzoFanAccount opened the pull request:
In version 0.1.41 the file tree nodes are seen as common elements rather than interactive ones so the agent incorrectly reports that files were downloaded even though none were.

Proposed fix:
In the DOM helper function `isInteractiveElement` , add a check for file tree nodes so that these elements are now treated as interactive and downloadable.

Fixes #1433 <!-- This is an auto-generated description by mrge. -->
---


## Summary by mrge
File tree nodes are now correctly recognized as interactive elements, so the agent no longer reports false file downloads.

<!-- End of auto-generated description by mrge. -->

pirate commented:
What is a `File Tree Node`? is this a class you see on elements from a specific website or JS library? Can you provide an example URL or HTML snippet?

EnzoFanAccount commented:
Yeah, your concern is valid. I'll reply to the original issue asking for the original url, a screenshot of the html for one of the file/folder items, and whatever else is useful

EnzoFanAccount commented:
@pirate Based on what the user provided, I've made a handful of changes to the PR.
added `treeitem` to `interactiveRoles`: This is the primary fix to recognize the list items in the Kendo UI TreeView based on their semantic role.
centralized Disable Check: Created a helper `isElementDisabled` to check disabled, readonly, inert, specific attributes, and non-interactive cursors. This check is now performed first. If an element is disabled, it's immediately marked as non-interactive, simplifying the following checks.
The checks are ordered from generally more reliable/semantic to less reliable:
Explicitly Disabled (definite non-interaction)
Standard HTML Tags (strong signal)
ARIA Roles (strong semantic signal)
`isContentEditable` (definite interaction)
Cursor Style (good heuristic, especially `pointer`)
Specific Heuristics (classes/attributes, used sparingly)
Event Listeners (fallback, least reliable)

Also I simplified the class/attribute checks to be less specific to frameworks unless absolutely necessary, focusing on `aria-haspopup` as a more general indicator.

Finally, I refined the listener check and focused it primarily on `click` events, as other events (`mousedown`, etc.) are less definitive indicators of the intended primary interaction.

Without the original URL I can't test it against the real thing, but it *should* in theory be a general fix to the bug. Give this a check when you get the chance

pirate commented:
@EnzoFanAccount pretty much every line in `browser_use/dom/buildDomTree.js` is written in blood, there's a very good reason for most of the checks there (and their original ordering).

We are open to improving it, but we cant make this many changes to the core algorithm for the sake of improving a single site without very extensive testing. (for context we spend $200+ on eval compute for basically every single-line changes to this file to confirm we're not introducing regressions)

If you'd like to open a new PR for a few specific changes at a time, it's more likely to get merged. For element detection PRs that affect multiple sites we basically require several reproducible examples w/ URLs and screenrecordings showing exactly what it improves on which sites and how.
